pyface/color.py

```diff
@@ -0,0 +1,260 @@
+
+    Returns
+    -------
+    channels : iterable of float
+        A tuple of channel values, each value between 0.0 and 1.0,
+        inclusive.
+    """
+    return tuple(value / maximum for value in values)
+
+
+#: A trait holding a single channel value.
+Channel = Range(0.0, 1.0, value=1.0, channel=True)
+
+#: A trait holding three channel values.
+ChannelTuple = Tuple(Channel, Channel, Channel)
+
+#: A trait holding four channel values.
+AlphaChannelTuple = Tuple(Channel, Channel, Channel, Channel)
+
+
+class Color(HasStrictTraits):
+    """ A mutable specification of a color with alpha.
+
+    This is a class designed to be used by user interface elements which
+    need to color some or all of the interface element.  Each color has a
+    number of different representations as channel tuples, each channel
+    holding a value between 0.0 and 1.0, inclusive.  The standard red,
+    green, blue and alpha channels are also provided as convenience
+    properties.
+
+    Methods are provided to convert to and from toolkit-specific color
+    objects.
+
+    Colors implement equality testing, but are not hashable as they are
+    mutable, and so are not suitable for use as dictionary keys.  If you
+    need a dictionary key, use an appropriate channel tuple from the
+    object.
+    """
+
+    #: A tuple holding the red, green, blue, and alpha channels.
+    rgba = AlphaChannelTuple()
+
+    #: A tuple holding the red, green, and blue channels.
+    rgb = Property(ChannelTuple(), depends_on='rgba')
+
+    #: The red channel.
+    red = Property(Channel, depends_on='rgba')
+
+    #: The green channel.
+    green = Property(Channel, depends_on='rgba')
+
+    #: The blue channel.
+    blue = Property(Channel, depends_on='rgba')
+
+    #: The alpha channel.
+    alpha = Property(Channel, depends_on='rgba')
+
+    #: A tuple holding the hue, saturation, value, and alpha channels.
+    hsva = Property(AlphaChannelTuple, depends_on=['rgba'])
+
+    #: A tuple holding the hue, saturation, and value channels.
+    hsv = Property(ChannelTuple, depends_on=['rgb'])
+
+    #: A tuple holding the hue, lightness, saturation, and alpha channels.
+    hlsa = Property(AlphaChannelTuple, depends_on=['rgba'])
+
+    #: A tuple holding the hue, lightness, and saturation channels.
+    hls = Property(ChannelTuple, depends_on=['rgb'])
+
+    @classmethod
+    def from_toolkit(cls, toolkit_color, **traits):
+        """ Create a new Color object from a toolkit color object.
+
+        Parameters
+        ----------
+        toolkit_color : toolkit object
+            A toolkit color object, such as a Qt QColor or a Wx wx.Colour.
+        **traits
+            Any additional trait values to be passed as keyword arguments.
+        """
+        from pyface.toolkit import toolkit_object
+        toolkit_color_to_rgba = toolkit_object('color:toolkit_color_to_rgba')
+        rgba = toolkit_color_to_rgba(toolkit_color)
+        return cls(rgba=rgba, **traits)
```

Comment:
Q: I'm not sure why we're allowing users to pass `**traits` here.

Reply:
This is basic defensive programming: if we were ever to subclass `Color` in a way that added new traits (or worse, a subclass which multiply inherits), we want the ability to pass values for those additional traits into any factory class methods we might have.

It's the same reason that you should always have `**traits` args in `__init__` methods of `HasTraits` subclasses

Reply:
that's what i was wondering. resolving.